libtelemetry/src/main/java/com/mapbox/services/android/telemetry/TelemetryService.java

```diff
@@ -167,4 +167,14 @@ TelemetryService obtainService() {
       return TelemetryService.this;
     }
   }
+
+  public void updateLocationPriority(int priority) {
+    locationPriority = priority;
+
+    if (locationEngine != null) {
+      locationEngine.deactivate();
```

Comment:
If I remember correctly this is not enough to reset the priority of the location engine and I guess we need to call `LocationEngine#removeLocationUpdates` explicitly so it behaves properly.
`locationEngine.activate()` works well because we have a _onConnected_ callback in which we're calling `LocationEngine#requestLocationUpdates` but that's not the case for disconnecting (_onDisconnected_ callback doesn't exist).
That being said, we should test this thoroughly.

In any case, we should revisit the whole `LocationEngine` APIs 👀 https://github.com/mapbox/mapbox-java/issues/403